### ISSUE TITLE
Ideal file format: Layout information stored as structured information instead of text blob you have to compile

### ISSUE THREAD

**Issue by madig** [daltonmaag/file-format-discussion#5]
I'm currently trying to write a script to infer Unicode values for glyphs that have none (think `one` and variants `one.tnum` and a localised `one-hb`). Inferring the value for `one.tnum` is usually easy, but what if `one-hb` is only accessible via a `GSUB` `locl` substitution? Currently, you have to compile a UFO with it's feature file and then go through the final `GSUB` table and map out the substitutions.

It would be nice if an ideal format would be able to store comprehensive layout data in a structured way so you don't have to compile code (features) to answer the question.

Possible inspirations: VOLT, FontForge?

**Comment by belluzj** [comment]
If the new format has feature files somewhere (which it will probably have), then the data you're after will be a consequence of that feature file, so it shouldn't be stored again in the format, because that would duplicate the information

**Comment by madig** [comment]
Yes. On the other hand, kerning information is already stored in the UFO format, so ufo2ft has code to ignore that if there's a `kern` table in the feature.fea. You don't have to go for 100% coverage to make interesting information accessible to easy processing.

Discussion over at the ufo-spec tracker: https://github.com/unified-font-object/ufo-spec/issues/87#issuecomment-489240747

**Comment by madig** [comment]
@adrientetar is writing up thoughts about how to store OTL data in a structured way: https://gist.github.com/adrientetar/023296f2399c40c539776dcd86c097f9
